ui/build.gradle.kts

```diff
@@ -34,7 +34,10 @@ plugins {
 
 val openapi by configurations.creating { description = "Used to reference OpenAPI spec files" }
 
-dependencies { openapi(project(":nessie-model", "openapi")) }
+// Use  Nessie REST API V1 in UI for now. TODO: Migrate to current Nessie REST API from `model`
+dependencies { openapi(create("org.projectnessie", "nessie-model", "0.40.2").setTransitive(false)) }
+// Resolve the openapi dependency immediately to avoid accidental override during integration tests
+openapi.resolve()
```

Comment:
@snazy : This PR is not ready for review yet, but I'm wondering if you have any suggestions for this particular build script change - without `resolve` integration test would fail because they pulled the `current` Nessie model jar (e.g. 0.41.1-SNAPSHOT`).

Reply:
`.resolve` is a no go. If you're going to change the UI anyway, you can get away with the following for now.
That Download task isn't awesome, because it will perform a HTTP HEAD everytime, but maybe good enough as a _temporary_ workaround.

```
diff --git a/ui/build.gradle.kts b/ui/build.gradle.kts
index 14e610b1f..2874d9795 100644
--- a/ui/build.gradle.kts
+++ b/ui/build.gradle.kts
@@ -19,6 +19,7 @@ import com.github.gradle.node.npm.task.NpmInstallTask
 import com.github.gradle.node.npm.task.NpmSetupTask
 import com.github.gradle.node.npm.task.NpmTask
 import com.github.gradle.node.task.NodeSetupTask
+import fi.linuxbox.gradle.download.Download
 
 plugins {
   // The ':nessie-ui' module is technically not a Java library, but declaring it as such provides
@@ -30,15 +31,9 @@ plugins {
   signing
   id("com.github.node-gradle.node")
   `nessie-conventions`
+  id("fi.linuxbox.download") version "0.6"
 }
 
-val openapi by configurations.creating { description = "Used to reference OpenAPI spec files" }
-
-// Use  Nessie REST API V1 in UI for now. TODO: Migrate to current Nessie REST API from `model`
-dependencies { openapi(create("org.projectnessie", "nessie-model", "0.40.2").setTransitive(false)) }
-// Resolve the openapi dependency immediately to avoid accidental override during integration tests
-openapi.resolve()
-
 node {
   download.set(true)
   version.set("16.14.2")
@@ -67,17 +62,11 @@ val clean =
   }
 
 val pullOpenApiSpec by
-  tasks.registering(Sync::class) {
-    destinationDir = openApiSpecDir
-    from(project.objects.property(Configuration::class).value(openapi))
-
-    doLast {
-      copy {
-        openApiSpecDir.listFiles { f -> f.name.endsWith(".jar") }?.forEach { f -> from(zipTree(f)) }
-        into(openApiSpecDir)
-        include("META-INF/openapi/openapi.yaml")
-      }
-    }
+  tasks.registering(Download::class) {
+    val downloadDir = openApiSpecDir.resolve("META-INF/openapi")
+    val v1Openapi = downloadDir.resolve("openapi.yaml")
+    from("https://github.com/projectnessie/nessie/releases/download/nessie-0.42.0/nessie-openapi-0.42.0.yaml")
+    to(v1Openapi)
   }
 
 val nodeSetup =
```

Reply:
My thinking was to do UI changes in a follow-up PR. Will switch to the download task for now.

Reply:
Alternative approach: #5408

Reply:
implemented differently in #5408